### PR TITLE
Setup doctest and fix broken code examples raised

### PR DESCRIPTION
**Pull request by sadielbartholomew**
Quick PR to setup doctest for cfunits and make appropriate amendments to any code examples as raised by corresponding test failures.

With much thanks to @kinow who provided demonstrated how to do this with cf-python (see NCAS-CMS/cf-python#57) and whose code was largely lifted and shifted here. I thought I'd start with cfunits as a much smaller library to set this up as a POC so that we know how doctest works and any quirks it has, so that can gradually chip away at the docstrings in cfdm and then cf-python to make sure they each set themselves up appropriately (as in NCAS-CMS/cf-python#61) and can therefore be doctested too.

Raising as a draft initially since there are still two doctest failure cases, both relating to `AttributeError`s when trying to set `units` on a `Units` object. I imagine that is deprecated functionality but I just want to check before amending those docstrings.

### CODE REVIEW

File: cfunits/units.py [hunk 1]
```diff
@@ -2044,11 +2048,14 @@ def conform(cls, x, from_units, to_units, inplace=False):
     >>> Units.conform(a, Units('minute'), Units('second'))
     array([   0.,   60.,  120.,  180.,  240.])
     >>> print(a)
-    [ 0.  1.  2.  3.  4.]
-
-    >>> Units.conform(a,
-                      Units('days since 2000-12-1'),
-                      Units('days since 2001-1-1'), inplace=True)
+    [0. 1. 2. 3. 4.]
```
Comment (sadielbartholomew) [inline cfunits/units.py:2051]: I anticipate some problems with setting up doctest in a practical way for cfdm and cf-python with respect to array representation.

Namely, this case is a one-liner (as bad as it gets here in cfunits) but for higher-dimensional arrays in the other libraries the print output could easily be ugly in terms of alignment and whitespaces, and doctest doesn't seem to be able (or maybe more accurately want to try to by scope) to work out that `[0. 1. 2. 3. 4.]` is equal to `[ 0.  1.  2.  3.  4.]`, though I can get ti to ignore whitespace outside of an array `repr` by adding an option flag `doctest.NORMALIZE_WHITESPACE` to the doctest tests.

**Review by sadielbartholomew — COMMENTED** [review]
@davidhassell before I change the remaining docstrings to finalise this PR, I just want to check my plan is sensible regarding fixing the remaining doctest failures (two compound cases), all directly or indirectly due to being unable to set `units` on a `Units` object, i.e:

```pytb
File "/home/sadie/cfunits/cfunits/units.py", line 555, in cfunits.units.Units
Failed example:
    v.units = 's'
Exception raised:
    Traceback (most recent call last):
      File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest cfunits.units.Units[14]>", line 1, in <module>
        v.units = 's'
    AttributeError: can't set attribute
```

or similarly:

```pytb
File "/home/sadie/cfunits/cfunits/units.py", line 516, in cfunits.units.Units
Failed example:
    u.units = 'days since 2004-3-1'
Exception raised:
    Traceback (most recent call last):
      File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest cfunits.units.Units[2]>", line 1, in <module>
        u.units = 'days since 2004-3-1'
    AttributeError: can't set attribute
```

I suppose there used to be a `@setter` defined for `units` but at some point that was removed (can't see anything obvious in the changelog about it though)? In which case I can just remove all examples attempting to set `units` as the fix? I thought at first maybe there was a flag that could toggle the ability to set that on and off, but can't find one so assume not.

**Comment by sadielbartholomew** [comment]
(Comments above were discussed offline. We decided the examples should be removed.)

**Comment by sadielbartholomew** [comment]
(Force pushed to sort out the merge conflicts locally where they are much easier to manage in emacs. Tests pass locally, now re-triggering the CI via open-close.)

**Comment by sadielbartholomew** [comment]
(Force pushed to sort out the merge conflicts locally where they are much easier to manage in emacs. Tests pass locally, now re-triggering the CI via open-close.)

**Review by kinow — APPROVED** [review]
:clap:

**Comment by sadielbartholomew** [comment]
Thanks @kinow: `doctest` is great it! :tada: Here it has picked up on a number of deprecated examples and some missing quote marks and the like! `cfunits` is a small library so it wasn't time consuming to get all examples passing with `doctest`, but I'll do another of our libraries, `cfdm`, next by incremental process since there are many, many more docstrings, a lot of which need some logic to set up objects they refer to.

**Comment by sadielbartholomew** [comment]
(Merging as checks have all passed locally and on the CI.)

**Comment by kinow** [comment]
It is really great! I have an old Java project with javadocs that contain examples too. Recently found several were broken/outdated, and wished there was something as good as doctest for Java.

**Pull request merged**